Maxxor.PCL/ValueObject/MxSouthAfricanIdNumber/MxSouthAfricanIdNumber.cs

```diff
@@ -0,0 +1,178 @@
+                : MxResult.Ok(this);
+        }
+
+        #region Date Of Birth
+
+        public MxDate DateOfBirth { get; private set; }
+        private MxResult PopulateDateOfBirth(string idNumberString)
+        {
+            var yearSubString = idNumberString.Substring(0, 2);
+            var monthSubString = idNumberString.Substring(2, 2);
+            var daySubString = idNumberString.Substring(4, 2);
+
+            try
+            {
+                var day = int.Parse(daySubString);
+                var month = int.Parse(monthSubString);
+                var year = int.Parse(yearSubString);
+
+                if (year > 0 && year < 11)
+                {
+                    year += 2000;
+                }
+                else
+                {
+                    year += 1900;
+                }
+
+                DateOfBirth = new MxDate(day, month, year);
+            }
+            catch (Exception e)
+            {
+                var fail = MxResult.Fail(this, MxSouthAfricanIdNumberError.InvalidDateOfBirth, e);
+                fail.Error.AddData("idNumber", idNumberString);
+                return fail;
+            }
+
+            return MxResult.Ok();
+        }
+
+        #endregion
+
+        #region Gender
+
+        public MxGender Gender { get; private set; }
+
+        private MxResult PopulateGender(string idNumberString)
+        {
+            var genderString = idNumberString.Substring(6, 4);
+            var genderNumber = int.Parse(genderString);
+
+            if (genderNumber >= 0 && genderNumber < 5000)
+            {
+                Gender = MxGender.Female;
+                return MxResult.Ok();
+            }
+            if (genderNumber >= 5000 && genderNumber <= 9999)
+            {
+                Gender = MxGender.Male;
+                return MxResult.Ok();
+            }
+
+            return MxResult.Fail(this, MxSouthAfricanIdNumberError.InvalidGender)
+                .AddData("idNumber", idNumberString); ;
+
+        }
+
+        #endregion
+
+        #region Citizenship
+
+        public bool IsCitizen { get; private set; }
+        public bool IsPermanentResident { get; private set; }
+
+        private MxResult PopulateCitizenship(string idNumberString)
+        {
+            var citizenshipSubstring = idNumberString.Substring(10, 1);
+
+            if (citizenshipSubstring == "0")
+            {
+                IsCitizen = true;
+                IsPermanentResident = false;
+            }
+            else
+            {
+                IsCitizen = false;
+                IsPermanentResident = true;
+            }
+
+            return MxResult.Ok();
+        }
+
+        #endregion
+
+        #region Checksum
+
+        private MxResult ValidateChecksum(string idNumberString)
+        {
+            var checksumString = idNumberString.Substring(12, 1);
+            var actualChecksum = int.Parse(checksumString);
```

Comment:
This `int.Parse` should be in a try-catch or you can use `int.TryParse` instead

Reply:
Done